spec/requests/api/v1/media_spec.rb

```diff
@@ -0,0 +1,191 @@
+      let(:media) { Fabricate(:media_attachment) }
+
+      it 'returns http not found' do
+        subject
+
+        expect(response).to have_http_status(404)
+      end
+    end
+
+    context 'when media is attached to a status' do
+      let(:media) { Fabricate(:media_attachment, account: user.account, status: Fabricate.build(:status)) }
+
+      it 'returns http not found' do
+        subject
+
+        expect(response).to have_http_status(404)
+      end
+    end
+  end
+
+  describe 'POST /api/v1/media' do
+    subject do
+      post '/api/v1/media', headers: headers, params: params
+    end
+
+    let(:params) { {} }
+
+    shared_examples 'a successful media upload' do |media_type|
+      it 'uploads the file successfully' do
+        subject
+
+        aggregate_failures do
+          expect(response).to have_http_status(200)
+          expect(MediaAttachment.first).to be_present
+          expect(MediaAttachment.first).to have_attached_file(:file)
+        end
+      end
+
+      it 'returns the correct media content' do
+        subject
+
+        body = body_as_json
+
+        expect(body).to match(
+          a_hash_including(id: MediaAttachment.first.id.to_s, description: params[:description], type: media_type)
+        )
+      end
+    end
+
+    it_behaves_like 'forbidden for wrong scope', 'read read:media'
+
+    describe 'when paperclip errors occur' do
+      let(:media_attachments) { double }
+      let(:params)            { { file: fixture_file_upload('attachment.jpg', 'image/jpeg') } }
+
+      before do
+        allow(User).to receive(:find).with(token.resource_owner_id).and_return(user)
+        allow(user.account).to receive(:media_attachments).and_return(media_attachments)
+      end
+
+      context 'when imagemagick cannot identify the file type' do
+        it 'returns http unprocessable entity' do
+          allow(media_attachments).to receive(:create!).and_raise(Paperclip::Errors::NotIdentifiedByImageMagickError)
+
+          subject
+
+          expect(response).to have_http_status(422)
+        end
+      end
+
+      context 'when there is a generic error' do
+        it 'returns http 500' do
+          allow(media_attachments).to receive(:create!).and_raise(Paperclip::Error)
+
+          subject
+
+          expect(response).to have_http_status(500)
+        end
+      end
+    end
+
+    context 'with image/jpeg' do
+      let(:params) { { file: fixture_file_upload('attachment.jpg', 'image/jpeg'), description: 'jpeg image' } }
+
+      it_behaves_like 'a successful media upload', 'image'
+    end
+
+    context 'with image/gif' do
+      let(:params) { { file: fixture_file_upload('attachment.gif', 'image/gif') } }
+
+      it_behaves_like 'a successful media upload', 'image'
+    end
+
+    context 'with video/webm' do
+      let(:params) { { file: fixture_file_upload('attachment.webm', 'video/webm') } }
+
+      it_behaves_like 'a successful media upload', 'image'
+    end
```

Comment:
I think the type here should be `video`. Am I missing something? 🤔

Reply:
Turns out I had to set `paperclip_processing: true` in the specs.

Reply:
Ah, good catch!

Reply:
I wonder if we should enable this for all the tests or maybe just “a successful media upload” tests. cc @mjankowski

Reply:
I'd say in general enable it for as narrow a slice as you can ... unless that's absurd (ie, 19/20 examples need it ... fine to just do for whole file).

In this case, I don't actually see what the 'successful media...' shared example is coming from?

Reply:
It's coming from https://github.com/mastodon/mastodon/pull/25543/files#diff-344de3dbcbeab7fb91612d4ebb4b212a37598d4868e46e075daab954789313ffL36-L74 although it's slightly more specific. I think it makes sense to test the end-to-end behavior of these endpoints.

Reply:
Ah, I see ... yeah, I think if enabled JUST for the shared context is the only place that needs it and it can stay turned off for the access control checks, thats slightly better.